### ISSUE TITLE
Comparison to pydantic-settings

### ISSUE THREAD

**Issue by sebastianneubauer** [Zuehlke/ConfZ#87]
Hi,
I am a bit puzzled right now which approach to use to get started and I somehow fail to see the big differences to the "official" pydantic-settings (https://github.com/pydantic/pydantic-settings/tree/main). What I do notice is:
* more "natural" semantics, as this is configuration, not "settings"
* ability to use config files, also with nested configuration and this is absolutlely a "killer feature" to me

The rest looks rather equal to me, am I missing something?

Is there more which justifies to pull in ConfZ as an additional dependency? 
Would it maybe even make sense to contribute the files source loader to pydantic-settings?

To summarize: Great project! ConfZ is exactly what I am looking for, but pydantic-settings is just missing a little bit and is the "official" recommendation from pydantic and fastapi. 

(sorry if this is explained already somewhere, then I just haven't found it)

**Comment by silvanmelchior** [comment]
Hi,

Thanks for your question!

ConfZ brings the following things with it, which are not provided by pydantic-settings:
- Loading config not only from env-vars, but also files and command-line arguments (and a combination of all of this, and easily extendable for more)
- Lazy-loading and caching options to load your config file once only and access it from everywhere
- Changing of config within context managers for e.g. unit tests

I think whether this is enough to justify another dependency depends on your project. If you want to work in a well-established stack with libraries supported by hundrets of developers, then ConfZ is probably not the right choice for you, whereas if you have a more dynamic environment you might want to give it a try ;).

**Comment by sebastianneubauer** [comment]
Thanks, that helps a lot!
